example/lib/main.dart

```diff
@@ -0,0 +1,259 @@
+import 'package:flutter/material.dart';
+import 'package:supabase_flutter/supabase_flutter.dart';
+
+Future<void> main() async {
+  await Supabase.initialize(
+      url: 'https://ualrvfiametspbkkntpo.supabase.co',
+      anonKey:
+          'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InVhbHJ2ZmlhbWV0c3Bia2tudHBvIiwicm9sZSI6ImFub24iLCJpYXQiOjE2NTk1ODU0MDIsImV4cCI6MTk3NTE2MTQwMn0.sMZmPO3ZKRK2-4J6SBc2k9ymoGYQPY7DxyYXxychX8I');
+  runApp(const MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  const MyApp({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      title: 'Supabase Flutter Demo',
+      home: MyWidget(),
+    );
+  }
+}
+
+class MyWidget extends StatefulWidget {
+  const MyWidget({Key? key}) : super(key: key);
+
+  @override
+  State<MyWidget> createState() => _MyWidgetState();
+}
+
+class _MyWidgetState extends State<MyWidget> {
+  User? _user;
+  @override
+  void initState() {
+    _getAuth();
+    super.initState();
+  }
+
+  Future<void> _getAuth() async {
+    setState(() {
+      _user = Supabase.instance.client.auth.currentUser;
+    });
+    Supabase.instance.client.auth.onAuthStateChange((event, session) {
```

Comment:
We should store the stream subscription and cancel it on dispose to avoid leaks.

Also, shouldn't we use SupabaseAuth? https://github.com/supabase-community/supabase-flutter/issues/170

Reply:
Thanks, I always forget this 😂

About using SupabaseAuth, I left a comment here that I would appreciate it a lot if you could take a look!
https://github.com/supabase-community/supabase-flutter/issues/170#issuecomment-1208985193